File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/adapter/SampleRecyclerViewType.java
package com.jdroid.android.sample.ui.adapter;

import android.view.View;
import android.widget.TextView;

import com.jdroid.android.recycler.RecyclerViewType;
import com.jdroid.android.sample.R;
import com.jdroid.android.sample.database.room.SampleEntity;

import androidx.recyclerview.widget.RecyclerView;

public abstract class SampleRecyclerViewType extends RecyclerViewType<SampleEntity, SampleRecyclerViewType.SampleHolder> {

	@Override
	protected Class<SampleEntity> getItemClass() {
		return SampleEntity.class;
	}

	@Override
	protected Integer getLayoutResourceId() {
		return R.layout.jdroid_default_item;
	}

	@Override
	public RecyclerView.ViewHolder createViewHolderFromView(View view) {
		SampleHolder holder = new SampleHolder(view);
		holder.name = findView(view, com.jdroid.android.R.id.name);
		return holder;
	}

	@Override
	public void fillHolderFromItem(SampleEntity item, SampleHolder holder) {
		holder.name.setText(item.getField());
	}

	public static class SampleHolder extends RecyclerView.ViewHolder {

		protected TextView name;

		public SampleHolder(View itemView) {
			super(itemView);
		}
	}
}

File: jdroid-android-sample/src/main/java/com/jdroid/android/sample/ui/adapter/SampleRecyclerViewType.kt
package com.jdroid.android.sample.ui.adapter

import android.view.View
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.jdroid.android.recycler.RecyclerViewType
import com.jdroid.android.sample.R
import com.jdroid.android.sample.database.room.SampleEntity

abstract class SampleRecyclerViewType : RecyclerViewType<SampleEntity, SampleRecyclerViewType.SampleHolder>() {

    override fun getItemClass(): Class<SampleEntity> {
        return SampleEntity::class.java
    }

    override fun getLayoutResourceId(): Int {
        return R.layout.jdroid_default_item
    }

    override fun createViewHolderFromView(view: View): RecyclerView.ViewHolder {
        val holder = SampleHolder(view)
        holder.name = findView(view, com.jdroid.android.R.id.name)
        return holder
    }

    override fun fillHolderFromItem(item: SampleEntity, holder: SampleHolder) {
        holder.name.text = item.field
    }

    class SampleHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        lateinit var name: TextView
    }
}